src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -777,6 +776,76 @@ public int EnsureCapacity(int capacity)
             return newSize;
         }
 
+        /// <summary>
+        /// Sets the capacity of this dictionary to what it would be if it had been originally initialized with all its entries
+        /// 
+        /// This method can be used to minimize the memory overhead 
+        /// once it is known that no new elements will be added. 
+        /// 
+        /// To completely clear a dictionary and release all memory 
```

Comment:
Nit: This sequence will not release all memory referenced by the dictionary. It will allocate minimum size storage array.